TodoApp/TodoApp.Api.Tests/Controllers/ItemListController.test.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+using NUnit.Framework;
+using TodoApp.Api.Controllers;
+using TodoApp.Api.Models;
+using TodoApp.Api.Tests.Comparers;
+
+namespace TodoApp.Api.Tests.Controllers
+{
+    internal static class ControllerExtensions
+    {
+        private static async Task<HttpStatusCode> ResolveResult(IHttpActionResult result)
+        {
+            var message = await result.ExecuteAsync(CancellationToken.None);
+            var actualStatusCode = message.StatusCode;
+            return actualStatusCode;
+        }
+
+        public static async Task<(HttpStatusCode, TResult)> ResolveResult<TController,TResult>(this TController controller, Func<TController,Task<IHttpActionResult>> actionSelector)
+        {
+            var result = await actionSelector(controller);
+            var message = await result.ExecuteAsync(CancellationToken.None);
+            var actualStatusCode = await ResolveResult(result);
+            message.TryGetContentValue(out TResult actualItems);
+            return (actualStatusCode, actualItems);
+        }
+
+        public static async Task<HttpStatusCode> ResolveResult<TController>(this TController controller, Func<TController, Task<IHttpActionResult>> actionSelector)
+        {
+            var result = await actionSelector(controller);
+            var actualStatusCode = await ResolveResult(result);
+            return actualStatusCode;
+        }
+    }
+
+    [TestFixture]
+    public class ItemListControllerTest
+    {
+        private ItemListController _controller;
+        
+        [SetUp]
+        public void SetUp()
+        {
+            _controller = new ItemListController
+            {
+                Request = new HttpRequestMessage(),
+                Configuration = new HttpConfiguration()
+            };
+        }
+
+        [Test]
+        public async Task GetAllItems_AllItemsReturned()
+        {
+            var expectedItems = new[]
+            {
+                new ItemModel {Id = Guid.Parse("c5cc89a0-ab8d-4328-9000-3da679ec02d3"), Text = "Make a cofee"},
```

Comment:
It was not necessary to change your coffee habits here 😆 However, since you did, you have a typo in `cofee` now 👼

Reply:
Oh, right! Since the third coffee is not cofffee anymore, it doesn't make sense to have cofee as the first one :)